support/tarball/tool-bootstrapping/bootstrap.sh

```diff
@@ -0,0 +1,231 @@
+    echo "      3. Build final sdk using final ref-pkgs and stage1 sdk and arcade"
+    echo ""
+}
+
+function LogMessage {
+    local size=(`df -h .`)
+    #local message="[${size[10]} - ${size[11]}] : $1"
+    local message=$1
+    local dt=`date '+%d/%m/%Y %H:%M:%S'`
+    local outFile="$bootstrapDir/status.log"
+
+    echo "==================================================================================================="
+    echo "= $message "
+    echo "==================================================================================================="
+
+    echo "[$dt] : $message" >> "$outFile"
+}
+
+function BuildStage1Sdk {
+    LogMessage "Starting Step1 - Build stage1 sdk" 
+
+    if [ -d "$stage1Dir" ]; then
+        echo "$stage1Dir already exists"
+        echo "exiting..."
+        exit 1
+    fi
+
+    if [ ! -d "$tarballSourceDir" ]; then
+        echo "$tarballSourceDir doesn't exist"
+        echo "exiting..."
+        exit 1
+    fi
+
+    # Copy tarball-source to stage1 directory
+    LogMessage "Copy tarball-source to stage1 directory"
+    cp -r $tarballSourceDir $stage1Dir
+
+    # Build stage1 source
+    LogMessage "Building stage1"
+    cd $stage1Dir
+    ./build.sh
+    cd $bootstrapDir
+
+    LogMessage "Completing Step1 - Build stage1 sdk" 
+}
+
+function BuildRefPkgs {
+    LogMessage "Starting Step2 - Build Reference Packages" 
+
+    if [ ! -d "$refPkgSourceDir" ]; then
+        echo "$refPkgSourceDir doesn't exist"
+        echo "exiting..."
+        exit 1
+    fi
+
+    if [ -d "$refPkgsDir" ]; then
+        echo "$refPkgsDir already exists"
+        echo "exiting..."
+        exit 1
+    fi
+
+    if [ ! -d "$stage1Dir" ]; then
+        echo "$stage1Dir doesn't exist"
+        echo "exiting..."
+        exit 1
+    fi
+
+    LogMessage "Copy reference-packages source to ref-pkgs dir"
+    cp -r $refPkgSourceDir $refPkgsDir
+
+    LogMessage "Copy coreclr-tools to ref-pkgs source-built dir"
+    cp -r $stage1Dir/Tools/source-built/coreclr-tools/ $refPkgsDir/source-built
+
+    LogMessage "Un-tar private source-built artifacts into ref-pkgs source-built dir"
+    tar -xzf $stage1Dir/bin/x64/Release/Private.SourceBuilt.Artifacts.0.1.0*.tar.gz --directory $refPkgsDir/source-built/
+
+    LogMessage "Un-tar sdk into ref-pkgs .dotnet dir"
+    if [ -d "$refPkgsDir/.dotnet/" ]; then
+        rm -rf $refPkgsDir/.dotnet/
+    fi
+    mkdir $refPkgsDir/.dotnet/
+    tar -xzf $stage1Dir/bin/x64/Release/dotnet-sdk-3*.tar.gz --directory $refPkgsDir/.dotnet/
+
+    LogMessage "Update SDK version in global.json file"
+    sdkVer=`ls $refPkgsDir/.dotnet/sdk`
+    LogMessage "    Sdk version = [$sdkVer]"
+    sed -i "s|\"dotnet\": \".*|\"dotnet\": \"$sdkVer\"|g" $refPkgsDir/global.json
+
+    LogMessage "Build stage1 ref-pkgs"
+    cd $refPkgsDir
+    ./build.sh
+
+    LogMessage "Package built reference packages into new tarball"
+    cd $refPkgsDir/artifacts/reference-packages
```

Comment:
Suggestion: end with the current directory either the same as what we started with, or a consistent current directory across all three steps.  It can get confusing if the script leaves you somewhere you didn't expect.

Reply:
I think after the call to the function, it is left in the same dir that it was in when the function was called.  But, I changed to pushd/popd just to be clear.